cycode/cli/printers/base_table_printer.py

```diff
@@ -0,0 +1,112 @@
+import abc
+from typing import List, NamedTuple, Dict, Optional
+
+import click
+from texttable import Texttable
+
+from cycode.cli.consts import SECRET_SCAN_TYPE, INFRA_CONFIGURATION_SCAN_TYPE, SAST_SCAN_TYPE
+from cycode.cli.printers.text_printer import TextPrinter
+from cycode.cli.models import DocumentDetections, CliError, CliResult
+from cycode.cli.printers.base_printer import BasePrinter
+
+
+class ColumnInfo(NamedTuple):
+    name: str
+    index: int  # lower index means left column
+    width_secret: int = 1
+    width_sast: int = 1
+    width_iac: int = 1
+
+
+class Table:
+    """Helper class to manage columns and their values in the right order and only if the column should be presented."""
+
+    def __init__(self, scan_type: str, columns_info: Optional[List[ColumnInfo]] = None):
+        self.scan_type = scan_type
+
+        self._columns: Dict[ColumnInfo, List[str]] = dict()
+        if columns_info:
+            self._columns: Dict[ColumnInfo, List[str]] = {columns: list() for columns in columns_info}
+
+    def add(self, column: ColumnInfo):
+        self._columns[column] = list()
+
+    def set(self, column: ColumnInfo, value: str):
+        # we pash values only for existing columns what were added before
+        if column in self._columns:
+            self._columns[column].append(value)
+
+    def _get_ordered_columns(self) -> List[ColumnInfo]:
+        # we are sorting columns by index to make sure that columns will be printed in the right order
+        return sorted(self._columns, key=lambda column_info: column_info.index)
+
+    def get_columns_info(self) -> List[ColumnInfo]:
+        return self._get_ordered_columns()
+
+    def get_headers(self) -> List[str]:
+        return [header.name for header in self._get_ordered_columns()]
+
+    def get_rows(self) -> List[str]:
+        ordered_values = [self._columns[column_info] for column_info in self._get_ordered_columns()]
+        return list(zip(*ordered_values))
```

Comment:
all the lists have the same length? if not we might omit lines when using zip. for example
[1,2,3] , [a,b,c], [!,@] -> [(1,a,!), (2,b,@)

Reply:
Yeap. If not textable also will raise the exception. Blank cell means empty sting means array item